core-cart.mjs

```diff
@@ -5,6 +5,75 @@ import * as os from "os";
 
 const discountedPrice = (price, rate = 0.10) => (price * (1 - rate)).toFixed(2);
 
+const createDelimiter = (openClose, symbol, times) => `${openClose} ${symbol.repeat(times)} ${openClose}`;
+
+const printShopName = () => 
+    `${os.userInfo().username.toUpperCase()} - Cart ${(process.pid)}`;
+
+
+const getUser = (uuid) => (users.find(user => user.uuid === uuid));
+
+const getData = () => {
+    let formatData = { weekday: 'short', year: 'numeric', month: 'short', day: 'numeric' };
+    let data = new Date();
+
+    return data.toLocaleString(`it-IT`, formatData);
+}
+
+const formatProductList = (lista) => {
+    let listaProdotti = ``;
+    lista.forEach(codice => {        
+        let prodotto = products.find(product => product.ean === codice);
+        listaProdotti += prodotto.ean 
+        + `\t` 
+        + prodotto.name + `\t`
+        + prodotto.price + `\t\t\t`;
+    });
+    return listaProdotti;
+}
+
+const formatProductName = (product) => product;
+
+const AlterType = (products, type) => products;
+
+const sumCartItem = (lista) => {
+    let somma = 0;
+    lista.forEach(codice => {
+        let prodotto = products.find(product => product.ean === codice);
+        somma += prodotto.price;
+    });
+    return somma.toFixed(2);
+}
+
+const printReceipt = (testo, contatore) => {
+    fs.writeFile(`./receipt/scontrino${contatore}.txt`, testo, err => {
```

Comment:
Leggi bene le richieste dentro il file readme manca qualcosa.

Reply:
Si purtroppo non sono riuscito bene a implementare tutte le richieste non avendole assimilate molto bene :(

Reply:
la cartella di atterraggio è `receipts` non `receipt` come menzionato nel README

Il nome del file doveva essere : `$UserUUID_receipt_$DATAORAGENERAZIONE.txt` dove `$UserUUID` è lo UUID dell'utente e $DATAORAGENERAZIONE è la data.

invece di chiamare: `core.printReceipt(scontrino, contatore);` (in cart.mjs) avresti dovuto chiamare `core.printReceipt(scontrino, cartUserUUID);`

Se riesci prova a modificare questo metodo per avere il nome corretto.